app/src/combo.c

```diff
@@ -262,37 +260,34 @@ static int clear_candidates() {
 }
 
 static int capture_pressed_key(const zmk_event_t *ev) {
-    for (int i = 0; i < CONFIG_ZMK_COMBO_MAX_KEYS_PER_COMBO; i++) {
-        if (pressed_keys[i] != NULL) {
-            continue;
-        }
-        pressed_keys[i] = ev;
-        return ZMK_EV_EVENT_CAPTURED;
+    if (pressed_keys_count == CONFIG_ZMK_COMBO_MAX_COMBOS_PER_KEY) {
+        return ZMK_EV_EVENT_BUBBLE;
     }
-    return ZMK_EV_EVENT_BUBBLE;
+
+    memcpy(&pressed_keys[pressed_keys_count++], ev,
+           sizeof(struct zmk_position_state_changed_event));
+    return ZMK_EV_EVENT_CAPTURED;
 }
 
 const struct zmk_listener zmk_listener_combo;
 
 static int release_pressed_keys() {
-    for (int i = 0; i < CONFIG_ZMK_COMBO_MAX_KEYS_PER_COMBO; i++) {
-        const zmk_event_t *captured_event = pressed_keys[i];
-        if (pressed_keys[i] == NULL) {
-            return i;
-        }
-        pressed_keys[i] = NULL;
+    uint32_t count = pressed_keys_count;
+    pressed_keys_count = 0;
+    for (int i = 0; i < count; i++) {
+        struct zmk_position_state_changed_event ev = pressed_keys[i];
```

Comment:
Make variable a pointer to avoid copy.

Reply:
Not yet changed to a pointer.

Reply:
So not sure about this, since the only use is to pass this to `ZMK_EVENT_RELEASE` or `ZMK_EVENT_RAISE` which expect an item on the stack.

Reply:
j/k. Fixed.